### PR TITLE
Bill controller test

### PR DESCRIPTION
**Pull request by jeysbach**
Fixed the existing api tests, added bill deletion and added a test for that as well.

### CODE REVIEW

File: test/e2e/api/BillController.test.ts [hunk 1]
```diff
@@ -62,17 +58,22 @@ test('create bill', async () => {
...
-    const createdBill = plainToClass(Bill, await res.json());
+    const createdId = (await res.text()).replace(/"/g, "");
+    const getBill = await fetch('http://localhost:' + process.env.PORT + '/api/bills/' + createdId, {
+        method: 'get',
+         headers: {
+            cookie: bearer,
+        },
+    });
+    const createdBill = plainToClass(Bill, await getBill.json());
     expect(createdBill.totalXrpDrops).toBe("100");
```
Comment (astamagg) [inline test/e2e/api/BillController.test.ts]: This line throws an error when I run the test, should be a number

**Review by astamagg — COMMENTED** [review]
I do realise that it is a part of an older accepted test but register.test.ts fails on the headers for register and login being different from what it expected

**Comment by jeysbach** [comment]
Both tests should be fixed now, no idea how the second one slipped through though 😅

**Review by astamagg — APPROVED** [review]
No worries, it happens to the best of us 😄 everything works as expected

**Pull request merged**